trie/trie.go

```diff
@@ -410,11 +410,16 @@ func (t *Trie) resolve(n node, prefix, suffix []byte) (node, error) {
 }
 
 func (t *Trie) resolveHash(n hashNode, prefix, suffix []byte) (node, error) {
-	if v, ok := globalCache.Get(n); ok {
-		return v, nil
+	var v node = nil
+	var ok bool
+	if v, ok = globalCache.Get(n); !ok {
+		enc, err := t.db.Get(n)
+		if err == nil && enc != nil {
+			v = mustDecodeNode(n, enc)
+		}
+		globalCache.Put(n, v)
```

Comment:
I think `globalCache.Put(hash, _)` must be called whenever `db.Put(hash, _)` happens.  Since `globalCache` now remembers cache misses, it must be informed when these cache misses do not happen anymore.